glib/glib.go

```diff
@@ -313,26 +314,54 @@ const (
 
 type SourceHandle uint
 
+// sourceFunc is the callback for g_idle_add_full and g_timeout_add_full that
+// replaces the GClosure API.
+//
+//export sourceFunc
+func sourceFunc(data C.gpointer) C.gboolean {
+	v := callback.Get(uintptr(data))
+	fs := v.(closure.FuncStack)
+
+	rv := fs.Func.Call(nil)
+	if len(rv) == 1 && rv[0].Bool() {
+		return C.TRUE
+	}
+
+	return C.FALSE
+}
+
+//export removeSourceFunc
+func removeSourceFunc(data C.gpointer) {
+	callback.Delete(uintptr(data))
+}
+
+var (
+	_sourceFunc       = (*[0]byte)(C.sourceFunc)
+	_removeSourceFunc = (*[0]byte)(C.removeSourceFunc)
+)
+
 // IdleAdd adds an idle source to the default main event loop context with the
 // DefaultIdle priority. If f is not a function with no parameter, then IdleAdd
 // will panic.
 //
 // After running once, the source func will be removed from the main event loop,
 // unless f returns a single bool true.
 func IdleAdd(f interface{}) SourceHandle {
-	return sourceAttach(
-		C.g_idle_source_new(),
-		f, PRIORITY_DEFAULT_IDLE, false,
-	)
+	return idleAdd(PRIORITY_DEFAULT, f)
```

Comment:
Is the change to PRIORITY_DEFAULT intended here?

Reply:
Very likely not. It seems that was a mistake on my end.

Reply:
I've made #776 to track this.